actors/miner/src/state.rs

```diff
@@ -877,46 +858,42 @@ impl State {
                 amount_unlocked
             ));
         }
-        // add locked funds now
-        vesting_funds.add_locked_funds(current_epoch, vesting_sum, self.proving_period_start, spec);
-        self.locked_funds += vesting_sum;
 
-        // save the updated vesting table state
-        self.save_vesting_funds(store, &vesting_funds)?;
+        // Finally, record the new locked-funds total.
+        self.locked_funds += vesting_sum;
 
         Ok(amount_unlocked)
     }
 
     /// Draws from vesting table and unlocked funds to repay up to the fee debt.
-    /// Returns the amount unlocked from the vesting table and the amount taken from
-    /// current balance. If the fee debt exceeds the total amount available for repayment
-    /// the fee debt field is updated to track the remaining debt.  Otherwise it is set to zero.
+    /// Returns the amount to burn and the total amount unlocked from the vesting table (both vested
+    /// and unvested) If the fee debt exceeds the total amount available for repayment the fee debt
+    /// field is updated to track the remaining debt. Otherwise it is set to zero.
     pub fn repay_partial_debt_in_priority_order<BS: Blockstore>(
         &mut self,
         store: &BS,
         current_epoch: ChainEpoch,
         curr_balance: &TokenAmount,
     ) -> Result<
         (
-            TokenAmount, // from vesting
-            TokenAmount, // from balance
+            TokenAmount, // fee to burn
+            TokenAmount, // total unlocked
         ),
         anyhow::Error,
     > {
-        let unlocked_balance = self.get_unlocked_balance(curr_balance)?;
-
         let fee_debt = self.fee_debt.clone();
-        let from_vesting = self.unlock_unvested_funds(store, current_epoch, &fee_debt)?;
+        let (from_vesting, total_unlocked) =
+            self.unlock_vested_and_unvested_funds(store, current_epoch, &fee_debt)?;
 
         if from_vesting > self.fee_debt {
             return Err(anyhow!("should never unlock more than the debt we need to repay"));
         }
-        self.fee_debt -= &from_vesting;
 
-        let from_balance = cmp::min(&unlocked_balance, &self.fee_debt).clone();
-        self.fee_debt -= &from_balance;
+        let unlocked_balance = self.get_unlocked_balance(curr_balance)?;
+        let to_burn = cmp::min(&unlocked_balance, &self.fee_debt).clone();
```

Comment:
I'm missing something here - why aren't we updating `&self.fee_debt` before calling this? we've already collected what we can `from_vesting` and we don't even use it, I don't see it impacting anything else in here so where is it going?

Reply:
We haven't collected anything yet. Basically, we:

1. Unlock up-to the fee debt from vesting rewards.
2. Then we pay the debt from the unlocked balance (includes both vested and unvested funds).

Reply:
ah, right, the coordination is through `self.locked_balance`
makes me want more inline docs tbh